game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1534,15 +1544,20 @@ public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
           }
         });
     if (firstStrikeOrder.getDefender() == FirstStrikeStepOrder.DEFENDER_NO_SNEAK_ATTACK) {
-      steps.add(
-          new FirstStrikeDefendersFire() {
-            private static final long serialVersionUID = 999921L;
+      steps.add(defensiveFirstStrike);
+      new IExecutable() {
+        private static final long serialVersionUID = 999921L;
 
-            @Override
-            public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-              firstStrikeDefendersFire(firstStrikeOrder.getDefender().getReturnFire());
-            }
-          });
+        @Override
+        public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+          new DefensiveFirstStrike(
+                  MustFightBattle.this,
```

Comment:
Seeing the same parameter twice is a bit difficult to grok. Have you considered using a builder for this constructor? In that case the parameters would be named and then more self-documenting.

Reply:
Alternatively, have you considered perhaps having a second constructor that is simplified, or using static builder methods (factory pattern)?

Reply:
The parameters are only used twice because currently the BattleAction and BattleState is tied to the same class.

I plan to have a factory class that creates all of the steps, sorts them by order, and then returns the items.  I can't use that yet because the steps aren't all converted so they would be out of order.  The backwards compatible classes, though, will always be a little weird.